### PR TITLE
Switch back to Carthage

### PR DESCRIPTION
**Pull request by frederoni**
- [x] Switch back to Carthage for development
- [x] Add an example project to ensure installing through CocoaPods doesn't regress

### CODE REVIEW

**Comment by frederoni** [comment]
@1ec5 @bsudekum Can I get some 👀 here. Also, take a look at the overhaul workflow in bitrise.

File: Cartfile [hunk 1]
```diff
@@ -3,4 +3,5 @@ github "mapbox/MapboxDirections.swift" ~> 0.8
 github "mapbox/MapboxGeocoder.swift" ~> 0.6
 github "52inc/Pulley" ~> 1.3.1
 github "rs/SDWebImage" ~> 4.0
-github "Project-OSRM/osrm-text-instructions.swift" ~> 0.0.1
+github "Project-OSRM/osrm-text-instructions.swift" "master"
```
Comment (bsudekum) [inline Cartfile]: Let's use a commit until a new release is out.

File: MapboxNavigationTests/CocoaPodsTest/PodInstall/Podfile [hunk 2]
```diff
@@ -0,0 +1,9 @@
+platform :ios, '9.0'
+use_frameworks!
+
+target 'PodInstall' do
+  pod 'MapboxGeocoder.swift', '~> 0.6.0'
+  pod 'MapboxNavigationUI.swift', :git => 'https://github.com/mapbox/MapboxNavigation.swift.git', :branch => 'BRANCH_NAME'
+  pod 'MapboxNavigation.swift', :git => 'https://github.com/mapbox/MapboxNavigation.swift.git', :branch => 'BRANCH_NAME'
+  pod 'OSRMTextInstructions', :git => 'https://github.com/Project-OSRM/osrm-text-instructions.swift', :branch => 'master'
```
Comment (bsudekum) [inline MapboxNavigationTests/CocoaPodsTest/PodInstall/Podfile]: Same thing as above, commit instead of master

Comment (bsudekum) [inline MapboxNavigationTests/CocoaPodsTest/PodInstall/Podfile]: Oh this is test file. Still I think all these branches should be moved to commits or releases.

Comment (frederoni) [inline MapboxNavigationTests/CocoaPodsTest/PodInstall/Podfile]: BRANCH_NAME is replaced in the workflow with $BITRISE_CURRENT_BRANCH but I agree on switching osrm-text-instructions to a specific release/commit.

File: MapboxNavigationTests/CocoaPodsTest/PodInstall/Podfile [hunk 3]
```diff
@@ -0,0 +1,9 @@
+platform :ios, '9.0'
+use_frameworks!
+
+target 'PodInstall' do
+  pod 'MapboxGeocoder.swift', '~> 0.6.0'
+  pod 'MapboxNavigationUI.swift', :git => 'https://github.com/mapbox/MapboxNavigation.swift.git', :branch => 'BRANCH_NAME'
+  pod 'MapboxNavigation.swift', :git => 'https://github.com/mapbox/MapboxNavigation.swift.git', :branch => 'BRANCH_NAME'
```
Comment (1ec5) [inline MapboxNavigationTests/CocoaPodsTest/PodInstall/Podfile]: This threw me for a loop. A comment to the effect of https://github.com/mapbox/MapboxNavigation.swift/pull/54#discussion_r103747252 would be useful here.

**Pull request merged**